File: 198. House Robber/houseRobber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int n = nums.size();
        if(n == 0) return 0;
        if(n <= 2) return max(nums[0],nums[n-1]);
        int last_3 = nums[0];
        int last_2 = nums[1];
        int last = nums[2] + last_3;
        for(int i = 3; i < n; ++i){
            int temp = nums[i] + max(last_2, last_3);
            last_3 = last_2;
            last_2 = last;
            last = temp;
        }
        return max(last, last_2);
    }
};

File: 198. House Robber/houseRobber.kt
class Solution {
    fun rob(nums: IntArray): Int {
        val n = nums.size
        if(n <= 2) return maxOf(nums[0], nums[n-1])
        var last_3 = nums[0]
        var last_2 = nums[1]
        var last = nums[0] + nums[2]
        for(i in 3..n-1){
            var temp = maxOf(last_2, last_3) + nums[i]
            last_3 = last_2
            last_2 = last
            last = temp
        }
        return maxOf(last, last_2)
    }
}
